packages/runtime-core/src/h.ts

```diff
@@ -84,6 +86,17 @@ export function h(
   children?: RawChildren | RawSlots
 ): VNode
 
+// text/comment
+export function h(
+  type: typeof Text | typeof Comment,
+  children?: RawChildren
+): VNode
+export function h(
+  type: typeof Text | typeof Comment,
+  props?: RawProps | null,
+  children?: RawChildren | RawSlots
+): VNode
+
```

Comment:
`Text` and `Comment` don't need `props`, and its children can only be of type `string`

```suggestion
export function h(
  type: typeof Text | typeof Comment,
  children?: string
): VNode
```

Reply:
- Neither does `Fragment`?
 - `h(Text, null, '')` is still valid

Reply:
```suggestion
// text/comment
export function h(
  type: typeof Text | typeof Comment,
  children?: string
): VNode
export function h(
  type: typeof Text | typeof Comment,
  props?: null,
  children?: string
): VNode
```